sklearn/metrics/_plot/tests/test_roc_curve_display.py

```diff
@@ -125,6 +125,79 @@ def test_roc_curve_display_plotting(
     assert display.ax_.get_xlabel() == expected_xlabel
 
 
+@pytest.mark.parametrize("plot_chance_level", [True, False])
+@pytest.mark.parametrize(
+    "chance_level_kwargs",
+    [
+        {"linewidth": 1},
+        {"color": "b", "label": "DummyEstimator"},
+        {"color": "g", "linestyle": ":"},
+    ],
+)
+@pytest.mark.parametrize(
+    "constructor_name",
+    ["from_estimator", "from_predictions"],
+)
+def test_roc_curve_chance_level_line(
+    pyplot,
+    data_binary,
+    plot_chance_level,
+    chance_level_kwargs,
+    constructor_name,
+):
+    """Check the chance leve line plotting behaviour."""
+    X, y = data_binary
+
+    lr = LogisticRegression()
+    lr.fit(X, y)
+
+    y_pred = getattr(lr, "predict_proba")(X)
+    y_pred = y_pred if y_pred.ndim == 1 else y_pred[:, 1]
+
+    if constructor_name == "from_estimator":
+        display = RocCurveDisplay.from_estimator(
+            lr,
+            X,
+            y,
+            alpha=0.8,
+            plot_chance_level=plot_chance_level,
+            chance_level_kwargs=chance_level_kwargs,
+        )
+    else:
+        display = RocCurveDisplay.from_predictions(
+            y,
+            y_pred,
+            alpha=0.8,
+            plot_chance_level=plot_chance_level,
+            chance_level_kwargs=chance_level_kwargs,
+        )
+
+    import matplotlib as mpl  # noqal
+
+    assert isinstance(display.line_, mpl.lines.Line2D)
+    assert display.line_.get_alpha() == 0.8
+    assert isinstance(display.ax_, mpl.axes.Axes)
+    assert isinstance(display.figure_, mpl.figure.Figure)
+
+    if plot_chance_level:
+        assert isinstance(display.chance_level_, mpl.lines.Line2D)
+        assert tuple(display.chance_level_.get_xdata()) == (0, 1)
+        assert tuple(display.chance_level_.get_ydata()) == (0, 1)
+
+        if "color" not in chance_level_kwargs:
+            assert display.chance_level_.get_color() == "k"
+        if "linestyle" not in chance_level_kwargs:
+            assert display.chance_level_.get_linestyle() == "--"
+        if "label" not in chance_level_kwargs:
+            assert display.chance_level_.get_label() == "Chance level (AUC = 0.5)"
```

Comment:
This will be simplified if we pass a single dict. We will not need for the `if` statement.

Reply:
Thanks for your review! I'm just a bit unsure about one last point. Do you mean I should do the following here:

```python
chance_level_line_kwargs = {
    "color": "k",
    "linestyle": "--",
    "label": "Chance level (AUC = 0.5)",
}
if chance_level_kwargs is not None:
    chance_level_line_kwargs.update(chance_level_kwargs)
for k, v in chance_level_line_kwargs.items():
    assert getattr(display.chance_level_, "get_" + k)() == v
```

Reply:
I commented next to the parametrization to clarify what I mean.